app/assets/javascripts/administration.js

```diff
@@ -0,0 +1,12 @@
+$(document).ready(function(){
+    $("#tabs li").click(function(){
+        if (!$(this).hasClass("active")) {
```

Comment:
Nope. This is just a matter of jQuery best practices. Each time you do `$(this)`, you're make jQuery search through the DOM to find that element. You can eliminate all that extra work by simply storing the jQuery object for that element in a variable. This pattern should be followed whenever you use the element more than once in your code. For example:

```
var $nav = $('#nav');

$nav.doSomething();
$nav.doSomethingElse();
```

The convention for identifying a variable that holds a jQuery object is to begin the name with `$`, which is ok to do since `$` has no special meaning in JavaScript. That way you eliminate confusion about what a variable is actually holding and whether or not it needs to be wrapped in `$()`. In addition, it's also a good idea to add classes to elements you'll be targeting with jQuery to separate the concerns of having classes for style and classes for logic. That convention is simply to append `js-` the name of the class.

```
<ul id="todoList" class="js-todoList">
    <li class="itemStyle active js-item"></li>
    <li class="itemStyle js-item"></li>
    <li class="itemStyle js-item"></li>
</ul>

var $this = $(this);
var $body = $('body');
var $todoList = $('#todoList');
// using .find() is helpful in reducing the scope of the lookup for large documents or if 
// a selector appears elsewhere in the page, but we only want items from a specific section
var $todoItems = $todoList.find('.js-item');
```

Now, just by looking at the markup, we can tell exactly what parts are going to be manipulated by jQuery. And our JavaScript is more efficient since we're storing references to the elements we need.